### PR TITLE
fix: show assets for available chains

### PR DESCRIPTION
**Pull request by Asmadek**
* check is chain id match for active accounts on assets page

### CODE REVIEW

**Comment by Asmadek** [comment]
Task linked: [CU-8693j5t6f Proxy. Assets page. Does not match with mockups](https://app.clickup.com/t/8693j5t6f)

File: src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx [hunk 1]
```diff
@@ -41,7 +42,9 @@ export const AssetsList = () => {
       const isDisabled = networkUtils.isDisabledConnection(connections[c.chainId]);
       const hasMultiPallet = !isMultisig || networkUtils.isMultisigSupported(c.options);
 
-      return !isDisabled && hasMultiPallet;
+      const hasChainAccount = activeAccounts.some((a) => accountUtils.isChainIdMatch(a, c.chainId));
```
Comment (tuul-wq) [inline src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx:45]: Suggest to create a map `{ [chainId]: true / false }` and check it inside `filteredChains` for each chain
For instance PV wallet with 300 accs would be looked up `chains x 300` times

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx [hunk 2]
```diff
@@ -38,11 +38,23 @@ export const AssetsList = () => {
...
+      ? new Set(Object.keys(chains) as ChainId[])
+      : activeAccounts.reduce<Set<ChainId>>((acc, account) => {
+          const chainId = (account as ChainAccount).chainId;
+
+          if (chainId) {
+            acc.add(chainId);
+          }
+
+          return acc;
+        }, new Set<ChainId>());
```
Comment (Tbaut) [inline src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx:51]: May I suggest to use a `filter` here for readability, something like
```suggestion
      : new Set(activeAccounts.filter((account) => !!account.chainId))
```

File: src/renderer/entities/wallet/lib/account-utils.ts [hunk 3]
```diff
@@ -57,8 +58,12 @@ function isAccountWithShards(accounts: Pick<Account, 'type'> | ShardAccount[]):
   return Array.isArray(accounts) && isShardAccount(accounts[0]);
 }
 
+function isAllChainsSupport(account: Pick<Account, 'type'>): boolean {
```
Comment (tuul-wq) [inline src/renderer/entities/wallet/lib/account-utils.ts:61]: I would suggest account-centric name like `isChainDependant`

```javascript
function isChainDependant(account: Pick<Account, 'type'>): boolean {
  return !isBaseAccount(account) && !isMultisigAccount(account);
}
```

File: src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx (hunk 2 above)
Comment (tuul-wq) [reply src/renderer/pages/Assets/Assets/components/AssetsList/AssetsList.tsx:51]: Thibaut's suggestion is nice and clean, `reduce` makes no calculations inside just filtering
```javascript
new Set(activeAccounts.filter((account) => Boolean(account.chainId)))
```

**Review by egor0798 — APPROVED** [review]

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**